skbio/metadata/tests/test_interval.py

```diff
@@ -446,6 +446,34 @@ def test_init_upper_bound_lt_lower_bound(self):
         with self.assertRaises(ValueError):
             IntervalMetadata(-1)
 
+    def test_upper_bound_is_none(self):
```

Comment:
Can you address this comment? `IntervalMetadata` defines an `__eq__`, so you can create an expected `IntervalMetadata` object and use `self.assertEqual` to compare it to the observed `IntervalMetadata` object. It is safer and shorter than the current implementation.